relay-server/src/services/buffer/envelope_stack/file_backed.rs

```diff
@@ -0,0 +1,624 @@
+use crate::envelope::{Envelope, EnvelopeError};
+use crate::services::buffer::common::ProjectKeyPair;
+use crate::services::buffer::envelope_stack::EnvelopeStack;
+use crate::services::buffer::envelope_store::file_backed::{
+    FileBackedEnvelopeStore, FileBackedEnvelopeStoreError,
+};
+use crate::statsd::{RelayCounters, RelayTimers};
+use hashbrown::HashSet;
+use std::error::Error;
+use std::io;
+use std::io::SeekFrom;
+use std::sync::{Arc, LazyLock};
+use tokio::fs::File;
+use tokio::io::{AsyncReadExt, AsyncSeekExt, AsyncWriteExt};
+use tokio::sync::Mutex;
+
+static ALLOWED_ERROR_KINDS: LazyLock<HashSet<io::ErrorKind>> = LazyLock::new(|| {
+    let mut error_kinds = HashSet::new();
+    error_kinds.insert(io::ErrorKind::InvalidData);
+    #[cfg(windows)]
+    error_kinds.insert(io::ErrorKind::Uncategorized);
+
+    error_kinds
+});
+
+/// The version of the envelope stack file format.
+const CURRENT_FILE_VERSION: u8 = 1;
```

Comment:
nit: We're in a module dealing with files.

```suggestion
const CURRENT_VERSION: u8 = 1;
```